src/process.cr

```diff
@@ -415,6 +415,27 @@ class Process
   private def close_io(io)
     io.close if io
   end
+
+  # Changes the root directory and the current working directory for the current
+  # process.
+  #
+  # ```
+  # Process.chroot("/var/empty")
+  # ```
+  def self.chroot(path : String) : Nil
+    path.check_no_null_byte
+    if LibC.chroot(path) != 0
+      raise Errno.new("Failed to chroot")
+    end
+
+    if LibC.chdir("/") != 0
+      errno = Errno.new("chdir after chroot failed")
+      errno.callstack = Callstack.new
```

Comment:
This doesn't compile, should be `CallStack`. My bad.

It'd be great if you could add some specs for this PR. You'll have to start a new process for testing chroot, but you can copy this: https://github.com/crystal-lang/crystal/blob/master/spec/std/kernel_spec.cr

Reply:
Thanks. I will have a look at this when I have a bit of spare time.